-fsanitize=address is not supported on illumos

**Describe the bug**
`LinuxlikeTests.test_c_link_args_and_env` and `LinuxlikeTests.test_env_cflags_ldflags` tests fails on illumos because there is no `-fsanitize=address` support.

<details>
<summary>Failure</summary>

```
___________________ LinuxlikeTests.test_c_link_args_and_env ____________________
[gw1] sunos5 -- Python 3.9.25 /usr/bin/python3.9

self = <unittests.linuxliketests.LinuxlikeTests testMethod=test_c_link_args_and_env>

    @skip_if_not_base_option('b_sanitize')
    def test_c_link_args_and_env(self):
        '''
        Test that the CFLAGS / CXXFLAGS environment variables are
        included on the linker command line when c_link_args is
        set but c_args is not.
        '''
        if is_cygwin():
            raise SkipTest('asan not available on Cygwin')
        if is_openbsd():
            raise SkipTest('-fsanitize=address is not supported on OpenBSD')

        testdir = os.path.join(self.common_test_dir, '1 trivial')
        env = {'CFLAGS': '-fsanitize=address'}
>       self.init(testdir, extra_args=['-Dc_link_args="-L/usr/lib"'],
                  override_envvars=env)
E       subprocess.CalledProcessError: Command '['/usr/bin/python3.9', '/data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/meson.py', 'setup', '--backend=ninja', '--prefix', '/usr', '--libdir', 'lib', '-Dc_link_args="-L/usr/lib"', '/tmp/tmp4q0co4jw', '/data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/test cases/common/1 trivial']' returned non-zero exit status 1.

unittests/linuxliketests.py:478: CalledProcessError
----------------------------- Captured stdout call -----------------------------
$ /usr/bin/python3.9 /data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/meson.py setup --backend=ninja --prefix /usr --libdir lib '-Dc_link_args="-L/usr/lib"' /tmp/tmp4q0co4jw '/data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/test cases/common/1 trivial'
stdout:
The Meson build system
Version: 1.11.1
Source dir: /data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/test cases/common/1 trivial
Build dir: /tmp/tmp4q0co4jw
Build type: native build
Project name: trivial test
Project version: undefined

test cases/common/1 trivial/meson.build:2:0: ERROR: Unable to detect linker for compiler `/usr/gcc/14/bin/gcc -Wl,--version -L/usr/lib -fsanitize=address`
stdout:
stderr: gcc: error: -fsanitize=address is not supported in this configuration


A full log can be found at /tmp/tmp4q0co4jw/meson-logs/meson-log.txt
NOTICE: You are using Python 3.9 which is EOL. Starting with v1.12.0, Meson will require Python 3.10 or newer

Build started at 2026-05-10T18:38:04.201274
Main binary: /usr/bin/python3.9
Build Options: -Dbackend=ninja -Dprefix=/usr -Dlibdir=lib '-Dc_link_args="-L/usr/lib"'
Python system: SunOS
The Meson build system
Version: 1.11.1
Source dir: /data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/test cases/common/1 trivial
Build dir: /tmp/tmp4q0co4jw
Build type: native build
Project name: trivial test
Project version: undefined
-----------
Detecting compiler via: `/usr/gcc/14/bin/gcc --version` -> 0
stdout:
gcc (OpenIndiana 14.3.0-oi-1) 14.3.0
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
-----------
Running command: -cpp -x c -E -dM -
-----
-----------
Detecting linker via: `/usr/gcc/14/bin/gcc -Wl,--version -L/usr/lib -fsanitize=address` -> 1
stderr:
gcc: error: -fsanitize=address is not supported in this configuration
-----------

test cases/common/1 trivial/meson.build:2:0: ERROR: Unable to detect linker for compiler `/usr/gcc/14/bin/gcc -Wl,--version -L/usr/lib -fsanitize=address`
stdout:
stderr: gcc: error: -fsanitize=address is not supported in this configuration


____________________ LinuxlikeTests.test_env_cflags_ldflags ____________________
[gw0] sunos5 -- Python 3.9.25 /usr/bin/python3.9

self = <unittests.linuxliketests.LinuxlikeTests testMethod=test_env_cflags_ldflags>

    @skip_if_not_base_option('b_sanitize')
    def test_env_cflags_ldflags(self):
        if is_cygwin():
            raise SkipTest('asan not available on Cygwin')
        if is_openbsd():
            raise SkipTest('-fsanitize=address is not supported on OpenBSD')

        testdir = os.path.join(self.common_test_dir, '1 trivial')
        env = {'CFLAGS': '-fsanitize=address', 'LDFLAGS': '-I.'}
>       self.init(testdir, override_envvars=env)
E       subprocess.CalledProcessError: Command '['/usr/bin/python3.9', '/data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/meson.py', 'setup', '--backend=ninja', '--prefix', '/usr', '--libdir', 'lib', '/tmp/tmpqiubfffg', '/data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/test cases/common/1 trivial']' returned non-zero exit status 1.

unittests/linuxliketests.py:1087: CalledProcessError
----------------------------- Captured stdout call -----------------------------
$ /usr/bin/python3.9 /data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/meson.py setup --backend=ninja --prefix /usr --libdir lib /tmp/tmpqiubfffg '/data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/test cases/common/1 trivial'
stdout:
The Meson build system
Version: 1.11.1
Source dir: /data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/test cases/common/1 trivial
Build dir: /tmp/tmpqiubfffg
Build type: native build
Project name: trivial test
Project version: undefined

test cases/common/1 trivial/meson.build:2:0: ERROR: Unable to detect linker for compiler `/usr/gcc/14/bin/gcc -Wl,--version -I. -fsanitize=address`
stdout:
stderr: gcc: error: -fsanitize=address is not supported in this configuration


A full log can be found at /tmp/tmpqiubfffg/meson-logs/meson-log.txt
NOTICE: You are using Python 3.9 which is EOL. Starting with v1.12.0, Meson will require Python 3.10 or newer

Build started at 2026-05-10T18:38:04.277960
Main binary: /usr/bin/python3.9
Build Options: -Dbackend=ninja -Dprefix=/usr -Dlibdir=lib
Python system: SunOS
The Meson build system
Version: 1.11.1
Source dir: /data/builds/ul-mass-rebuild/components/developer/meson/build/amd64-3.9/test cases/common/1 trivial
Build dir: /tmp/tmpqiubfffg
Build type: native build
Project name: trivial test
Project version: undefined
-----------
Detecting compiler via: `/usr/gcc/14/bin/gcc --version` -> 0
stdout:
gcc (OpenIndiana 14.3.0-oi-1) 14.3.0
Copyright (C) 2024 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
-----------
Running command: -cpp -x c -E -dM -
-----
-----------
Detecting linker via: `/usr/gcc/14/bin/gcc -Wl,--version -I. -fsanitize=address` -> 1
stderr:
gcc: error: -fsanitize=address is not supported in this configuration
-----------

test cases/common/1 trivial/meson.build:2:0: ERROR: Unable to detect linker for compiler `/usr/gcc/14/bin/gcc -Wl,--version -I. -fsanitize=address`
stdout:
stderr: gcc: error: -fsanitize=address is not supported in this configuration


```
</details>

**To Reproduce**
Run meson tests on illumos (e.g. OpenIndiana).

**Expected behavior**
Both tests either pass or skip.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)? native
* operating system: OpenIndiana
* Python version: 3.9.25, 3.14.4
* meson version: 1.11.1
* ninja version: 1.13.2

--- a/unittests/linuxliketests.py
+++ b/unittests/linuxliketests.py
@@ -472,6 +472,8 @@ class LinuxlikeTests(BasePlatformTests):
             raise SkipTest('asan not available on Cygwin')
         if is_openbsd():
             raise SkipTest('-fsanitize=address is not supported on OpenBSD')
+        if is_sunos():
+            raise SkipTest('-fsanitize=address is not supported on illumos')
 
         testdir = os.path.join(self.common_test_dir, '1 trivial')
         env = {'CFLAGS': '-fsanitize=address'}
@@ -1024,6 +1026,8 @@ class LinuxlikeTests(BasePlatformTests):
             raise SkipTest('asan not available on Cygwin')
         if is_openbsd():
             raise SkipTest('-fsanitize=address is not supported on OpenBSD')
+        if is_sunos():
+            raise SkipTest('-fsanitize=address is not supported on illumos')
 
         testdir = os.path.join(self.common_test_dir, '1 trivial')
         env = {'CFLAGS': '-fsanitize=address', 'LDFLAGS': '-I.'}

